src/Oceananigans.jl

```diff
@@ -132,7 +132,48 @@ end
 
 abstract type Architecture end
 struct CPU <: Architecture end
-struct GPU <: Architecture end
+
+# A slightly complex (but fully-featured?) implementation:
+
+struct Layout{NT, NB}

```

Comment:
`struct ThreadBlockLayout` would be much more descriptive I think.

Reply:
I think (but not 100% sure) that we will always want tuples of size 3 and just set dimensions we don't care about to 1 (or 0?).

So maybe we'll always have `NT=3` and `NB=3` but this is more flexible and nothing wrong with flexibility I suppose.

Reply:
Yes that's the thread-block layout I've been using. I think it could be improved, although when I benchmarked with a few different layouts it didn't seem to matter as long as you had lots of threads/blocks. So maybe having a more general layout helps with small problems.

But as part of resolving #64 I was going to use `NT=3` and `NB=3` so we can use the maximum number of threads on both small and large problems.

Might be worth discussing this in person as we might have to design a more general thread-block layout.

And I think the way we test that a thread-block layout is valid is via
```julia
Tx*Ty*Tz*Bx*By*Bz == Nx*Ny*Nz
```